crates/cli/src/subcommands/dns.rs

```diff
@@ -30,70 +30,27 @@ pub async fn exec(mut config: Config, args: &ArgMatches) -> Result<(), anyhow::E
-        .post(format!(
+        .put(format!(
             "{}/v1/database/{database_identity}/names",
             config.get_host_url(server)?
         ))
-        .body(String::from(domain));
+        .header(reqwest::header::CONTENT_TYPE, "application/json")
+        .body(serde_json::to_string(&[&domain])?);
     let builder = add_auth_header_opt(builder, &auth_header);
 
-    let result = builder.send().await?.json_or_error().await?;
-    match result {
-        InsertDomainResult::Success {
-            domain,
-            database_identity,
-        } => {
-            println!("Domain set to {} for identity {}.", domain, database_identity);
-        }
-        InsertDomainResult::TldNotRegistered { domain } => {
-            return Err(anyhow::anyhow!(
-                "The top level domain that you provided is not registered.\n\
-            This tld is not yet registered to any identity. You can register this domain with the following command:\n\
-            \n\
-            \tspacetime dns register-tld {}\n",
-                domain.tld()
-            ));
-        }
-        InsertDomainResult::PermissionDenied { domain } => {
-            //TODO(jdetter): Have a nice name generator here, instead of using some abstract characters
-            // we should perhaps generate fun names like 'green-fire-dragon' instead
-            let suggested_tld: String = identity.chars().take(12).collect();
-            if let Some(sub_domain) = domain.sub_domain() {
-                return Err(anyhow::anyhow!(
-                    "The top level domain {} is not registered to the identity you provided.\n\
-                We suggest you register a new tld:\n\
-                \tspacetime dns register-tld {}\n\
-                \n\
-                And then push to the domain that uses that tld:\n\
-                \tspacetime publish {}/{}\n",
-                    domain.tld(),
-                    suggested_tld,
-                    suggested_tld,
-                    sub_domain
-                ));
-            } else {
-                return Err(anyhow::anyhow!(
-                    "The top level domain {} is not registered to the identity you provided.\n\
-                We suggest you register a new tld:\n\
-                \tspacetime dns register-tld {}\n\
-                \n\
-                And then push to the domain that uses that tld:\n\
-                \tspacetime publish {}\n",
-                    domain.tld(),
-                    suggested_tld,
-                    suggested_tld
-                ));
-            }
-        }
-        InsertDomainResult::OtherError(e) => return Err(anyhow::anyhow!(e)),
+    let response = builder.send().await?;
+    let status = response.status();
+    let response_body = response.text().await?;
+    if !status.is_success() {
+        anyhow::bail!("{response_body}");
     }
 
+    println!("Domain set to {} for identity {}.", domain, database_identity);
```

Comment:
We should probably just say name here. We pulled out support for TLDs so domain is confusing.